### PR TITLE
Query builder MVP stabilization

### PR DESCRIPTION
**Pull request by akphi**
| Q                       | A                                |
| ----------------------- | -------------------------------- |
| Fixed issues?           |  |
| Tests added + passed?   | Yes                              |
| Changeset added?        | Yes                              |
| Documentation PR link   |                                  |
| Any dependency changes? | Yes                               |

This is more a continuation of query builder MVP in https://github.com/finos/legend-studio/pull/166

This adds support for more complex filter features and stabilizes the query builder UI. In particular, the major points of this PR are:
- [x] Handling property expressions which includes properties with multiplicity `many - [*]` in filter with `exists()`
- [x] Properly handling group operations with more than 2 clauses
- [x] Add the ability to simplify filter boolean expression tree (unnecessary group node can be squashed into its parent group node who has the same group operator). This is done to help with handling reading filter expression with group operations with more than 2 clauses
- [x] Add support for set operators, such as `in/not-in`
- [x] Fix issue where enumeration operators cannot resolve the enum values after exiting text-mode
- [x] Add more round-trip tests for the query filter lambda processing
- [x] Stabilize certain DnD interactions in filter boolean expression tree
- [x] Respect user's input for variable names in lambda
- [x] Add a workaround to hide `graph fetch` option in query builder as it's `WIP`

### CODE REVIEW

File: packages/legend-studio-plugin-query-builder/src/stores/QueryBuilderFilterState.ts [hunk 1]
```diff
@@ -676,23 +727,49 @@ export class QueryBuilderFilterState
...
+       * like and/or do not take more than 2 parameters, as such, if we have more than 2, we need
+       * to create a chain of this operation to accomondate.
+       *
+       * This means that in the read direction, we might need to flatten the chains down to group with
+       * multiple clauses. This means user's intended grouping will not be kept.
+       */
+      if (clauses.length > 2) {
+        const firstClause = clauses[0];
+        let currentClause: ValueSpecification = clauses[clauses.length - 1];
+        for (let i = clauses.length - 2; i > 0; --i) {
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-studio-plugin-query-builder/src/stores/QueryBuilderFilterState.ts:759]: no there isn't, as long as if I'm a group node and my parent group node is of same operator as mine, then I can be squashed

File: packages/legend-studio-plugin-query-builder/src/components/__tests__/QueryBuilder.test.tsx [hunk 2]
```diff
@@ -478,31 +560,21 @@ test(integrationTest('Test unsupported functions'), async () => {
...
-    "Can't find type 'model::pure::tests::model::simple::NotFound",
+  testRoundtrip(lambda_enumerationOperatorFilter, mockedEditorStore);
+  testRoundtrip(lambda_existsChainFilter, mockedEditorStore);
+  testRoundtrip(
+    lambda_existsChainFilterWithCustomVariableName,
+    mockedEditorStore,
   );
-  expect(() =>
-    queryBuilderState.buildWithRawLambda(getRawLambda(unSupportedFunctionName)),
-  ).toThrowError(`Can't process function 'testUnSupported'`);
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-studio-plugin-query-builder/src/components/__tests__/QueryBuilder.test.tsx:507]: a non-roundtrip test? Yeah I think I can try to design that.

File: .changeset/mean-emus-clean.md [hunk 3]
```diff
@@ -1,9 +1,8 @@
 ---
 '@finos/legend-studio': minor
-'@finos/legend-studio-components': minor
-'@finos/legend-studio-app': minor
 ---
 
+pr: #171
```
[the remark this answers is not in the document]
Comment (akphi) [reply .changeset/mean-emus-clean.md:5]: For this: https://github.com/atlassian/changesets/pull/535

I wanted to change the content of this changeset, but in the generated changelog, I don't want this changeset to be associated with this PR.

File: packages/legend-studio-plugin-query-builder/src/components/QueryBuilderValueSpecificationEditor.tsx [hunk 4]
```diff
@@ -148,10 +170,215 @@ const EnumValueInstanceValueEditor = observer(
...
+      }
+      default:
+        // unsupported expected type, just escape
+        return;
+    }
+  } else if (expectedType instanceof Enumeration) {
+    result = uniq(parseData.map((item) => item.trim()))
+      .map((item): EnumValueInstanceValue | undefined => {
+        const _enum = returnUndefOnError(() => expectedType.getValue(item));
+        if (!_enum) {
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-studio-plugin-query-builder/src/components/QueryBuilderValueSpecificationEditor.tsx:271]: Yeah the only reason why I didn't do something like a notification is that I don't want to pass `editorStore` to `ValueSpecEditor`, sounds to me like the latter can be more generic and be made a reusable components in many apps. But I will add a comment for sure.

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**